packages/rest/test/unit/request-body.validator.test.ts

```diff
@@ -201,13 +300,16 @@ describe('validateRequestBody', () => {
 
 function verifyValidationRejectsInputWithError(
   errorMatcher: Error | RegExp | string,
+  details: RestHttpErrors.ErrorDetails[] | undefined,
   body: object | null,
   spec: RequestBodyObject | undefined,
   schemas?: SchemasObject,
 ) {
-  // workaround for Function.prototype.bind not preserving argument types
-  function validateRequestBodyWithBoundArgs() {
+  try {
     validateRequestBody(body, spec, schemas);
+    throw new Error('function `validateRequestBody` should throw error.');
+  } catch (err) {
+    expect(err.message).to.equal(errorMatcher);
```

Comment:
we should use shouldjs's `throw` to assert for properties in the error object

Reply:
@shimks shoudjs doesn't support a customized error assertion function, that's why I have to take this workaround. I need to do a deep equal check for `error.details`.

And I tried do the assertion inside a `expect.to.throw`, like this:
```js
function shouldThrowError() {
  try {
    throwError();
  } catch (err) {
    // assertion 1
    expect(err.details).to.deepEqual(expectedDetails);
  }  
}

// assertion 2
expect(shouldThrowError).to.throw(expectedMessage)
```

Its problem is if assertion 1 fails with `message1`, assertion 2 will receive `message 1` instead of the `expectedMessage` thrown from `throwError()`, then assertion 2 fails too(but it shouldn't), and prints a very unreadable message, sth like:
```
// message 1 is a super long error log for a deepEqual assertion
message 1 doesn't match expectedMessage
```

Reply:
@jannyHou When you say shouldjs does not support customized error assertion, do you mean that it can't take in an additional object with key/value pairs that may exist inside the error object, or that it can't do deep assertions on the properties? Because `throw()` can take in a second argument with the customized properties: https://shouldjs.github.io/#assertion-throw

I'm trying to push for the `throw()` so that the code doesn't have to manually throw an error if no error was thrown. I guess it's not a big deal though

Reply:
@shimks thanks for the doc 👍 will try it if I have time but like Miroslav said, it probably couldn't do a deep equality check. will let you know then.

Reply:
@shimks even I provide `{details: details}` in the 2nd input like:
```ts
  function validate() {
    validateRequestBody(body, spec, schemas);
  }
  expect(validate).to.throw(errorMatcher, {
   // tests still pass if you change the code to `details: {}`
    details: details,
  });
```
it doesn't do the equality check, but only checks the error message match. Tests still pass if you change the code to `details: {}`.

Reply:
I think the correct usage would be a single-arg invoke of `throw`:

```ts
expect(validate).to.throw({
  message: /* message text (strict comparison? not ideal) */
  details: details
});
```

Please note that `{details: details}` can be shortened to `{details}`, and also don't forget to `await` the expect statement.

```ts
await expect(validate).to.throw({details});
// or 
await expect(validate).to.throw({
  message: /*...*/,
  details,
});
```

I don't really mind implementation details of `verifyValidationRejectsInputWithError` as long as it works correctly in all cases. As I wrote earlier, the current implementation is fine with me.